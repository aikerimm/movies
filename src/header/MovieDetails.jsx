import PropTypes from 'prop-types';

const MovieDetails = ({
  movie: { imageName, title, rating, runtime, genre, releaseDate, description },
}) => {
  return (
    <>
      <img src={imageName} alt='moviePoster' className='movieCardPoster' />
      <div className='movieDetails'>
        <div className='movieDetailsHorizonalDisplay'>
          <p className='titleText'>{title}</p>
          <div className='circle'>{rating}</div>
        </div>
        <p className='movieDetailsGenre'>{genre}</p>
        <div className='movieDetailsHorizonalDisplay'>
          <p className='redText'>{releaseDate.getFullYear()}</p>
          <p className='redText'>
            {Math.floor(runtime / 60)}h {runtime % 60}min
          </p>
        </div>
        <p>{description}</p>
      </div>
    </>
  );
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    imageName: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    rating: PropTypes.number,
    runtime: PropTypes.number.isRequired,
    genre: PropTypes.arrayOf(PropTypes.string).isRequired,
    releaseDate: PropTypes.instanceOf(Date).isRequired,
    description: PropTypes.string,
  }),
};

export default MovieDetails;
