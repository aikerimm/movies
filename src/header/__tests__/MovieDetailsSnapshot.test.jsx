import renderer from 'react-test-renderer';
import MovieDetails from '../MovieDetails.jsx';

it('MovieDetails renders correctly', () => {
  const movie = {
    title: 'Zootopia',
    vote_average: 9.8,
    runtime: 120,
    genres: ['Drama', 'Comedy'],
    release_date: '1994-01-01',
    overview: 'Lorem Ipsum Some',
    poster_path: '',
  };
  const tree = renderer.create(<MovieDetails movie={movie} />).toJSON();
  expect(tree).toMatchSnapshot();
});
