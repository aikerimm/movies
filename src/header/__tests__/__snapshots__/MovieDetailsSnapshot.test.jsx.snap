// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
[
  <img
    alt="moviePoster"
    className="movieCardPoster"
    src=""
  />,
  <div
    className="movieDetails"
  >
    <div
      className="movieDetailsHorizonalDisplay"
    >
      <p
        className="titleText"
        id="movieTitleInHeader"
      >
        Zootopia
      </p>
      <div
        className="circle"
      >
        9.8
      </div>
    </div>
    <p
      className="movieDetailsGenre"
    >
      Drama, Comedy
    </p>
    <div
      className="movieDetailsHorizonalDisplay"
    >
      <p
        className="redText"
      >
        1994
      </p>
      <p
        className="redText"
      >
        2
        h 
        0
        min
      </p>
    </div>
    <p>
      Lorem Ipsum Some
    </p>
  </div>,
]
`;
